src/csm_testing/tests/sat_functional/nid2xname/test_nid2xname.py

```diff
@@ -35,6 +35,22 @@ def run_command(self, command):
         result = subprocess.run(command, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         return result.stdout.decode().strip()
 
+    def get_consecutive_components(self, min_count=3):
+        """
+        Return a list of components with at least min_count consecutive NIDs.
+        """
+        command = (
+            "cray hsm state components list --type Node --format json | "
+            "jq -r '.Components | map({NID: .NID, Xname: .ID}) | unique | sort_by(.NID)'"
```

Comment:
Out of curiosity, did you run into an issue that made you add the "unique" filter in your `jq` command? I would think the list of nid, xname pairs would already be unique as both of those should already be unique in HSM.